### ISSUE TITLE
Sound delay (first keystroke)

### ISSUE THREAD

**Issue by TRaMeLL** [hainguyents13/mechvibes#13]
There is a delay in sound when you first press a key in a new application window.

Windows 10x64 \ Mechvibes 2.1.1

![SysSpec1](https://user-images.githubusercontent.com/10239461/85226316-7e9ac900-b409-11ea-94c1-ccc39bdb7b21.png)

**Comment by hainguyents13** [comment]
Does it delay when you first start Mechvibes or open any new app window?

**Comment by TRaMeLL** [comment]
Any new app window and after idle. And it looks like the problem was solved by rebooting PC.

**Comment by hainguyents13** [comment]
delay after idle is a known issue and it happens to all of the "keyboard events manager" apps
please understand

**Comment by TRaMeLL** [comment]
Yes, I understand, but maybe there is a way to prevent the delay. As I said, there is no delay if Mechvibes is not minimized to system tray.

**Comment by hainguyents13** [comment]
Thanks for the information, that's helpful. I'm working on that.

**Comment by botsblitz32** [comment]
Encountered this too, delay is more noticeable in Linux than in Windows

**Comment by hainguyents13** [comment]
I'm still working hard on that, and due to my lack of knowledge about javascript GB, it's harder than I thought.
If anyone has a solution, please let me know!

**Comment by l3afblow3r** [comment]
It appears that the application is sleeping as it sees that it doesn't receive any keystrokes.
Once you start typing the loop starts running again.
This is my idea of what might be happening...

**Comment by MathiasF1do** [comment]
this issue is still ongoing

**Comment by nekelpatrick** [comment]
Yep. Can confirm that the issue is still ongoing. Worked fine on Windows 10 & 11. Delay is awful on Mac M1 Sequoia 15.0

**Comment by hainguyents13** [comment]
From my understanding, this delay seems to be caused by limitations in JavaScript itself — likely due to how it handles heavy or frequent tasks (e.g. event loop, GC).
I’m working on a Rust version that may solve this completely, and it’s showing good results so far.
Closing this issue as it's not something I can feasibly fix in JavaScript. Thanks for reporting and stay tuned for updates on the Rust rewrite.

**Comment by FytikProdd** [comment]
Hello! If you are on Windows, we recently created an optimized fork called **Mechvibes X** which fixes audio latency/lag issues by cleaning up legacy bloat, optimization of the audio graph (bypasses panners when spatial audio is disabled), and addition of OS session shutdown handlers to ensure the application closes and releases key hooks cleanly. It also introduces true **3D Spatial Audio** and **Pitch/Volume Randomization**! You can download the portable executable here: https://github.com/FytikProdd/mechvibes-x
